go/vt/vtgate/planbuilder/testdata/union_cases.json

```diff
@@ -2287,7 +2287,8 @@
   {
     "comment": "union with invalid order by clause with table qualifier",
     "query": "select id from user union select 3 order by id",
-    "plan": "VT12001: unsupported: ORDER BY on top of UNION"
+    "v3-plan": "VT12001: unsupported: ORDER BY on top of UNION",
+    "gen4-plan": "VT13001: [BUG] ORDER BY in complex query *planbuilder.distinct"
```

Comment:
instead of bug it should be unsupported error message.

Reply:
I didn't quite figure out why this was happening. My thinking was that once we take on DISTINCT on the operators, we would figure out what's wrong.